### DISCUSSION TITLE
Mochawesome HTML report: Missing scenario steps (stack) in CodeceptJS 3 when scenario fails

### DISCUSSION THREAD

**Discussion by mirao** [codeceptjs/CodeceptJS#3028] [General]
#### What are you trying to achieve?
[Mochawesome](https://codecept.io/reports/#html) report shows scenario steps (stack) when a scenario fails. It's useful because of context - when it's not clear in which step it occurred. 
This is how it looks in CodeeceptJS 2.6.11:

![image](https://user-images.githubusercontent.com/12584138/102697287-0f295900-4235-11eb-85d5-da2814b292b7.png)

#### What do you get instead?
Missing scenario steps in CodeceptJS 3.0.3. There is only an "internal" stack from node modules which isn't too much useful:

![image](https://user-images.githubusercontent.com/12584138/102697354-7e06b200-4235-11eb-8b10-e3069d04f707.png)

> Provide console output if related. Use `--verbose` mode for more details.

```bash
# codeceptjs run --grep "Test login screen"  --profile jobr  --reporter mocha-multi
```

### Details

* CodeceptJS version: 3.0.3
* NodeJS Version: v12.20.0
* mocha-multi: 1.1.3
* mochawesome: 6.1.1
* Operating System: Ubuntu 20.04/20.10
* Configuration file:

```js
    mocha: {
        reporterOptions: {
            "codeceptjs-cli-reporter": {
                stdout: "-",
                options: {
                    verbose: false,
                    debug: true,
                    steps: true
                }
            },
            mochawesome: {
                stdout: "./output/console.log",
                options: {
                    reportDir: "./output",
                    reportFilename: "report"
                }
            },
            "mocha-junit-reporter": {
                stdout: "./output/console.log",
                options: {
                    mochaFile: "./output/result.xml",
                    jenkinsMode: true,
                    attachments: true //add screenshot for a failed test
                },
            }
        }
    },
```

**Comment by elukoyanov** [comment]
yep, I see this error too on 3.0.2

For failed tests there are error stack and recorder code 
![image](https://user-images.githubusercontent.com/11647141/102788314-7d942580-43b3-11eb-98c5-9b9763ee886e.png)

For successful tests - some code 
![image](https://user-images.githubusercontent.com/11647141/102788500-c0ee9400-43b3-11eb-897a-f91cd16900c2.png)

I am using WebDriver helper without gherkin steps.

Do you use gherkin, @mirao ?

**Comment by mirao** [comment]
@elukoyanov No, I do not. I'm also using WebDriver without gherkin

**Comment by ktryniszewski-mdsol** [comment]
I think I accidentally suppressed the gherkin error in 3.0.3 when i was adding logic for bddStep event. Instead of throwing the error it was returning on the whole step when error is encountered. I am not sure if this is related, but I have a fix for that here: https://github.com/codeceptjs/CodeceptJS/pull/2745

**Comment by kiranreddyk06** [comment]
Hi, I face the same issue. I can not see any steps/verbose information in the report. 

![image](https://user-images.githubusercontent.com/76172127/109032896-05e9bc00-76be-11eb-9993-d8b8804aa3b8.png)

**Comment by ShawnBaumbach** [comment]
I'm having the same issue, in the json file, context becomes null and the code is replaced with the same recorder.errHandler blob you posted.

Using the latest versions of mochawesome (6.2.2) and codeceptjs (3.0.6)

This only happens if you retry the test. The old data is replaced with this incorrect data.

Screenshots remain in the folder, but are also not displayed.

This was not a problem with 2.x.x

> "context": null,
>               "code": "recorder.errHandler((err) => {\n  recorder.session.start('teardown');\n  recorder.cleanAsyncErr();\n  if (test.throws) { // check that test should actually fail\n    try {\n      assertThrown(err, test.throws);\n      event.emit(event.test.passed, test);\n      event.emit(event.test.finished, test);\n      recorder.add(() => done());\n      return;\n    } catch (newErr) {\n      err = newErr;\n    }\n  }\n  event.emit(event.test.failed, test, err);\n  event.emit(event.test.finished, test);\n  recorder.add(() => done(err));\n});\nif (isAsyncFunction(testFn)) {\n  event.emit(event.test.started, test);\n  const catchError = e => {\n    recorder.throw(e);\n    recorder.catch((e) => {\n      const err = (recorder.getAsyncErr() === null) ? e : recorder.getAsyncErr();\n      recorder.session.start('teardown');\n      recorder.cleanAsyncErr();\n      event.emit(event.test.failed, test, err);\n      event.emit(event.test.finished, test);\n      recorder.add(() => done(err));\n    });\n  };\n  let injectedArguments;\n  try {\n    injectedArguments = getInjectedArguments(testFn, test);\n  } catch (e) {\n    catchError(e);\n    return;\n  }\n  testFn.call(test, injectedArguments).then(() => {\n    recorder.add('fire test.passed', () => {\n      event.emit(event.test.passed, test);\n      event.emit(event.test.finished, test);\n    });\n    recorder.add('finish test', () => done());\n    recorder.catch();\n  }).catch(catchError);\n  return;\n}\ntry {\n  event.emit(event.test.started, test);\n  testFn.call(test, getInjectedArguments(testFn, test));\n} catch (err) {\n  recorder.throw(err);\n} finally {\n  recorder.add('fire test.passed', () => {\n    event.emit(event.test.passed, test);\n    event.emit(event.test.finished, test);\n  });\n  recorder.add('finish test', () => done());\n  recorder.catch();\n}",
>               "err": {
>                 "message": "Error: Text \"We've sent your info to\" was not found on page after 10 sec\nwaiting for function failed: timeout 10000ms exceeded",
>                 "estack": "Error: Text \"We've sent your info to\" was not found on page after 10 sec\nwaiting for function failed: timeout 10000ms exceeded\n    at C:\\PLM\\auto-quote-and-buy-ui\\node_modules\\codeceptjs\\lib\\helper\\Puppeteer.js:2091:13\n    at async Actor.takeScreenshot (e2e\\src\\steps\\basic.steps.js:118:5)\n    at async Test.<anonymous> (e2e\\src\\tests\\features\\send-to-agent_test.js:43:3)",
>                 "diff": null
>               },

**Reply by elukoyanov** [reply]
still repoduces
3.1.3 + 6.2.3 and 3.2.3 + 7.0.1 with retries >=1

**Comment by rockwalrus** [comment]
Related issue: #3274
